### PR TITLE
Reviewer AMC: RealmManager should register peer hooks rather than the Diameter stack

### PR DESCRIPTION
**Pull request by graemerobertson**
Hi Andy,

Fixes https://github.com/Metaswitch/cpp-common/issues/299.

Only register for the peer connected/failed hook when we've actually got a realm manager.

I haven't tested this, it doesn't seem that controversial to me and I don't have time to test it at the moment - what do you think?

Thanks,
Graeme

### CODE REVIEW

File: src/diameterstack.cpp [hunk 1]
```diff
@@ -121,13 +115,27 @@ void Stack::initialize()
...
 }
 
+void Stack::register_peer_hook_hdlr()
+{
+  int rc = fd_hook_register(HOOK_MASK(HOOK_PEER_CONNECT_SUCCESS,
+                                      HOOK_PEER_CONNECT_FAILED),
+                            fd_peer_hook_cb,
+                            this,
+                            NULL,
+                            &_peer_cb_hdlr);
```
Comment (bossmc) [inline src/diameterstack.cpp]: Shouldn't the callback be provided by the `RealmManager` too?

Comment (bossmc) [inline src/diameterstack.cpp]: I know this gets cleaned up in `~Diameter::Stack()` but it should probably also be possible for the `RealmManager` to do it's own unregister for symmetry.

Comment (graemerobertson) [inline src/diameterstack.cpp]: I can't really envisage what you're suggesting, and I doubt you're going to convince me to implement it - it sounds like a lot of work for virtually no gain. Do you think it's required for this PR?

I'm happy to implement your second comment.

**Comment by bossmc** [comment]
Ok fine.

**Pull request merged**